x-pack/platform/packages/shared/ai-infra/inference-common/src/errors.ts

```diff
@@ -17,44 +18,22 @@ export enum InferenceTaskErrorCode {
   abortedError = 'requestAborted',
 }
 
-/**
- * Base class for all inference API errors.
- */
-export class InferenceTaskError<
+const InferenceTaskError = ServerSentEventError;
```

Comment:
@pgayvallet WDYT here? not very proud of this, but:

- translating InferenceTaskError from and to ServerSentEventError everywhere is a lot of work
- they're basically the same
- _not_ aliasing InferenceTaskError to ServerSentEventError is also a little weird, because essentially ServerSentEventError is an implementation detail and specific to SSE.